products/sentinel_one.py

```diff
@@ -153,15 +99,136 @@ def _authenticate(self):
+                if counter == 10 or i == len(account_ids) - 1:
+                    response = self._get_all_paginated_data(self._build_url(f'/web/api/v2.1/accounts'),
+                                                            params={'ids': ','.join(temp_list)},
+                                                            add_default_params=False)
+
+                    if 'errors' in response:
+                        if response['errors'][0]['code'] == 4010010:
+                            raise ValueError(f'Failed to authenticate to SentinelOne: {response}')
+                        else:
+                            raise ValueError(f'Error when authenticating to SentinelOne: {response}')
+
+                    for account in response:
+                        if account['id'] not in self._account_ids:
+                            self._account_ids.append(account['id'])
+
+                    counter = 0
+                    temp_list = []
+                i += 1
+
+            diff = list(set(account_ids) - set(self._account_ids))
+            if len(diff) > 0:
+                self.log.warning(f'Account IDs {",".join(diff)} not found.')
+
+        if account_names: # verify provided account names are valid
+            temp_account_name = list()
+            for name in account_names:
+                response = self._get_all_paginated_data(self._build_url('/web/api/v2.1/accounts'),
+                                                        params={'name': name},
+                                                        add_default_params=False)
+
+                if 'errors' in response:
+                    if response['errors'][0]['code'] == 4010010:
+                        raise ValueError(f'Failed to authenticate to SentinelOne: {response}')
+                    else:
+                        raise ValueError(f'Error when authenticating to SentinelOne: {response}')
+
+                for account in response:
+                    temp_account_name.append(account['name'])
+                    if account['id'] not in self._account_ids:
+                        self._account_ids.append(account['id'])
+            
+            diff = list(set(account_names) - set(temp_account_name))
+            if len(diff) > 0:
+                self.log.warning(f'Account names {",".join(diff)} not found')
+        
+        if site_ids: # ensure specified site IDs are valid and not already covered by the account_ids listed above
+            temp_site_ids = list()
+            # create batches of 10 site_ids
+            counter = 0
+            temp_list = []
+            i = 0
+            while i < len(site_ids):
+                temp_list.append(site_ids[i])
+                counter += 1
+                if counter == 10 or i == len(site_ids) - 1:
+                    response = self._get_all_paginated_data(self._build_url('/web/api/v2.1/sites'),
+                                                            params={'siteIds': ','.join(site_ids)},
```

Comment:
```suggestion
                                                            params={'state': "active", 'siteIds': ','.join(site_ids)},
```
only return active sites. any expired/deleted sites will show in mismatch check of temp_site_ids and site_ids on line 221.